src/helper/convertColumnElementsToData.js

```diff
@@ -99,4 +112,85 @@ function convertColumnElementsToData(childComponents) {
   };
 };
 
-export default convertColumnElementsToData;
+/**
+ *
+ */
+function collectPropsAndTemplatesFromColumnData(column, columnProps, elementTemplates) {
+  columnProps.push(_extractPropsFromData(column));
+  _extractTemplatesFromData(elementTemplates, column);
+}
+
+/**
+ * Uses columnGetter to fetch column props and cell rendering templates
+ *
+ * @param {!function(number)} columnGetter
+ * @param {!number} columnsCount
+ * @return {{
+ *   columnGroupProps: !Array.<columnDefinition>,
+ *   columnProps: !Array.<columnDefinition>,
+ *   elementTemplates: !Object.<string, Array>,
+ *   useGroupHeader: boolean,
+ * }}
+ */
+function fetchColumnData(columnGetter, columnsCount) {
```

Comment:
this is similar to what was done [before](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/helper/convertColumnElementsToData.js#L47) (the default behavior)

Instead of iterating over React.Children and then extracting and calculating column/column group props, we simply use `columnGetter` prop to fetch the column info. 

The `columnGetter` will call ask the user for a column by specifying the index and whether if it's grouped or not.